test/models/CommentModel/test.js

```diff
@@ -0,0 +1,84 @@
+const {
+  expect, CommentModel, faker, sinon, Mongoose,
+} = require('test/testHelper');
+const { CommentFactory } = require('test/factories');
+const _ = require('lodash');
+
+describe('Comment Model', async () => {
+  describe('On getOne', async () => {
+    let comment, commentCount, nameAuthor;
+    beforeEach(async () => {
+      commentCount = 6;
+      nameAuthor = faker.name.firstName();
+      for (let iteration = 0; iteration < commentCount; iteration++) {
+        if (iteration === 0) comment = await CommentFactory.Create({ author: nameAuthor });
+        else await CommentFactory.Create({});
+      }
+    });
+    it('Should check comment for identity using author key', async () => {
+      const result = await CommentModel.getOne({ ..._.pick(comment, 'author', 'permlink') });
+      expect(result.comment).to.deep.eq(comment);
+    });
+    it('Should check comment for identity using userId key ', async () => {
+      const result = await CommentModel.getOne({ ..._.pick(comment, 'permlink', 'userId') });
+      expect(result.comment).to.deep.eq(comment);
+    });
+    it('Should return null comment', async () => {
+      const result = await CommentModel.getOne({ author: faker.name.firstName() });
+      expect(result.comment).to.be.null;
+    });
+    it('Should check that the error exists', async () => {
+      sinon.stub(Mongoose.Model, 'findOne').throws('DataBase is not responding');
+      const result = await CommentModel.getOne({ author: faker.name.firstName() });
+      expect(result.error).to.be.exist;
```

Comment:
If you want use sinon, use it in new describe at beforeEach hook... and use afterEach hook where you must restore sinon (sinon.restore()), if you do not, then errors will occur. If you have only one it where you use sinon, you can use it in the IT case, but you must add afterEach hook with sinon restore (ask me, i will tell you about sinon restore)

Reply:
You must stub only Comment.findOne, not all Mongoose Model findOne methods